mimic/model/maas_objects.py

```diff
@@ -0,0 +1,229 @@
+"""
+MaaS API data model
+"""
+
+import random
+
+
+def random_letters(len_min, len_max):
+    """
+    Makes a string with random letters (a-z).
+    """
+    return ''.join([chr(ord('a') + random.randint(0, 25))
+                    for _ in xrange(random.randint(len_min, len_max))])
+
+
+class Metric(object):
+    """
+    Models a MaaS metric type.
+    """
+    def __init__(self, name, type, clock, unit='other'):
+        """
+        Initializes the metric.
+        """
+        self.name = name
+        self.type = type
+        self._unit = unit
+        self._overrides = {}
+        self._clock = clock
+
+    def _override_key(self, **kwargs):
+        """
+        Computes a key used for hashing overrides, as a tuple of strings.
+        """
+        return (kwargs['entity_id'],
+                kwargs['check_id'],
+                kwargs.get('monitoring_zone', '__AGENT__'))
+
+    def set_override(self, **kwargs):
+        """
+        Sets the override metric for a given tenant, entity and check.
+
+        Override metrics are defined as a function which takes a timestamp
+        and returns the metric value.
+        """
+        self._overrides[self._override_key(**kwargs)] = kwargs['override_fn']
+
+    def clear_overrides(self):
+        """
+        Clears the override metric values.
+        """
+        self._overrides = {}
+
+    def _get_default_data(self):
+        """
+        Gets the default data point. This data point may be overridden by
+        setting the override value.
+        """
+        if self.type == 'i':
```

Comment:
It would be helpful to those who are not as familiar with the MaaS API to hoist these magic codes (`'i'`, `'n'`, `'s'`) into named / documented constants.